File: core/src/main/java/com/cz/android/sample/processor/clazz/DialogFragmentClassActionProcessor.java
package com.cz.android.sample.processor.clazz;

import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.FragmentActivity;

import com.cz.android.sample.api.item.RegisterItem;

/**
 * @author Created by cz
 * @date 2020-01-27 15:41
 * @email bingo110@126.com
 */
public class DialogFragmentClassActionProcessor extends ClassActionProcessor{
    @Override
    public boolean isInstance(Class item) {
        return super.isInstance(item)&& DialogFragment.class.isInstance(item);
    }

    @Override
    public void run(FragmentActivity context, RegisterItem registerItem, Class clazz) throws InstantiationException, IllegalAccessException {
        DialogFragment dialog = (DialogFragment) clazz.newInstance();
        dialog.show(context.getSupportFragmentManager(),null);
    }
}

File: core/src/main/java/com/github/jackchen/android/core/processor/DialogFragmentClassActionProcessor.kt
package com.github.jackchen.android.core.processor

import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.DialogFragment
import com.github.jackchen.android.core.exception.SampleFailedException
import com.github.jackchen.android.sample.api.SampleItem

class DialogFragmentClassActionProcessor : ActionProcessor {

  override fun isAvailable(clazz: Class<*>): Boolean {
    return DialogFragment::class.java.isInstance(clazz)
  }

  @Throws(SampleFailedException::class)
  override fun execute(context: AppCompatActivity, sampleItem: SampleItem) {
    val dialog = sampleItem.clazz().getConstructor().newInstance() as DialogFragment
    dialog.show(context.supportFragmentManager, null)
  }
}
